research-software-directory/RSD-as-a-service
# Orcid link and case insensitive auth endpoint enabling

# Show ORCID and case insensitivity auth module

Changes proposed in this pull request:

* The auth module is now case insensitive when reading values from the env variable `RSD_AUTH_PROVIDERS`
* When a contributor or team member has an ORCID, we show it on the page and link to the ORCID page
* Enforce the ORCID structure in the database and frontend

How to test:
* Change the case of the values you have in the env variable `RSD_AUTH_PROVIDERS` (mixed case should also work)
* The database structure has changed, clear and build:
	* `docker-compose down --volumes && docker-compose build --parallel && docker-compose up --scale scrapers=0`
* Login and create a project and add a team member with an ORCID
* Check that the proper structure is enforced
* View the project, the ORCID should be visible and link to the ORCID page
* Do the same for a software entry


Closes #280
Closes #286

PR Checklist:
